### PR TITLE
[NFC] CRM-19068 clean up GroupNesting class

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
This removes some unused things from the code

@seamuslee001 I was finding the other commit a bit much to read so have put part of it in here

---
- [CRM-19068](https://issues.civicrm.org/jira/browse/CRM-19068)

### CODE REVIEW

File: CRM/Group/Form/Edit.php [hunk 1]
```diff
@@ -396,7 +396,6 @@ public function postProcess() {
       }
       $params['group_type'] = $groupTypeIds;
 
-      $customFields = CRM_Core_BAO_CustomField::getFields('Group');
```
Comment (eileenmcnaughton) [inline CRM/Group/Form/Edit.php]: This var is not used

File: CRM/Group/Form/Edit.php [hunk 2]
```diff
@@ -237,7 +237,7 @@ public function buildQuickForm() {
     );
 
     $groupTypes = CRM_Core_OptionGroup::values('group_type', TRUE);
-    $config = CRM_Core_Config::singleton();
```
Comment (eileenmcnaughton) [inline CRM/Group/Form/Edit.php]: This var is not used

**Comment by seamuslee001** [comment]
@eileenmcnaughton can you fix the style errors?

**Comment by eileenmcnaughton** [comment]
done - I think I removed all the functions!

**Comment by seamuslee001** [comment]
@eileenmcnaughton It seems the test failures are legit. I think the cache code needs to stay for some strange reason.

File: CRM/Contact/BAO/Group.php [hunk 3]
```diff
@@ -448,9 +443,6 @@ public static function create(&$params) {
         }
       }
 
-      // clear any descendant groups cache if exists
-      $finalGroups = CRM_Core_BAO_Cache::deleteGroup('descendant groups for an org');
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Group.php]: @seamuslee001 aha - it was the removal of this line causing all the issues - it looked pretty innoculous - removing a pointless delete but it was doing a cache clear as a hidden side-effect

**Comment by seamuslee001** [comment]
I've looked at this and feel its good to merge, with the tests passing i'm happy for this cleanup to be merged.

**Comment by eileenmcnaughton** [comment]
Yes!! nearly 600 points!

**Pull request merged**